service/vector_store_client.cc

```
@@ -261,9 +283,13 @@ struct vector_store_client::impl {
     std::function<future<std::optional<inet_address>>(sstring const&)> dns_resolver;
     sequential_producer<lw_shared_ptr<http_client>> client_producer;
 
-    impl(host_name host_, port_number port_)
-        : host(std::move(host_))
-        , port(port_)
+    impl(utils::config_file::named_value<sstring> cfg)
+        : uri_observer(cfg.observe([this](std::string_view uri) {
+            _host_port = get_host_port(uri);
+            trigger_dns_refresh();
+            vslogger.info("Vector Store service URI changed to '{}'", uri);
+        }))
+        , _host_port(get_host_port(cfg()))
```

We should handle uri values differently than in live update:

- Do the user want to disable vector_store_client explicitly using is_vector_store_disabled. We should log that.
- Do the user provide invalid uri? We should throw a configuration exception explicitly in the constructor, not in the called `get_host_port`.

---

Done.
* When uri is empty the following log appears: "Vector Store service URI is empty, disabling Vector Store service"
* When invalid uri is at the beginning the exception is throws - as before